timemachine/fe/atom_mapping.py

```diff
@@ -364,13 +373,16 @@ def chiral_filter(trial_core):
         filter_fxn=filter_fxn,
     )
 
-    if connected_core:
-        all_cores = remove_disconnected_components(mol_a, mol_b, all_cores, all_marcs)
+    all_bond_cores = [_compute_bond_cores(mol_a, mol_b, marcs) for marcs in all_marcs]
 
-    if complete_rings:
-        assert connected_core
-        all_cores, all_marcs = remove_incomplete_rings(mol_a, mol_b, all_marcs, all_cores)
-        all_cores = remove_disconnected_components(mol_a, mol_b, all_cores, all_marcs)
+    if connected_core and complete_rings:
+        all_cores, all_bond_cores = remove_disconnected_components(mol_a, mol_b, all_cores, all_bond_cores)
+        all_cores, all_bond_cores = remove_incomplete_rings(mol_a, mol_b, all_cores, all_bond_cores)
+        all_cores, all_bond_cores = remove_disconnected_components(mol_a, mol_b, all_cores, all_bond_cores)
```

Comment:
Non-obvious logic: Comment constraints on filter application order

Reply:
(Resolved in https://github.com/proteneer/timemachine/pull/920/commits/91d2acbfb08ae22bf1c964a941389c56f2582122 )